lib/MusicBrainz/Server/Controller/Account.pm

```diff
@@ -297,6 +297,13 @@ sub edit : Local RequireAuth DenyWhenReadonly {
             $form->value
         );
 
+        my $old_username = $editor->username;
+        my $new_username = $form->field('username')->value;
+              
+        if (defined $new_username && $new_username ne $old_username) {
+            $c->detach('/error_403');            
+        }
+
```

Comment:
Can't test this, because the account/edit form seems unrelatedly bugged (about languages) but thought I'd put it up so you could say if it looks reasonable or not at least.